public/main.js

```diff
@@ -80,4 +91,22 @@ function deleteEntry(todo) {
     createRequest.send();
 }
 
+function completeEntry(todo) {
+    var createRequest = new XMLHttpRequest();
+    createRequest.open("PUT", "/api/todo/");
+    createRequest.setRequestHeader("Content-type", "application/json");
+    createRequest.onload = function() {
```

Comment:
This onload() method is exactly the same as the one in deleteEntry(), but with a different error message. Best to refactor this into a method that takes in an error message as the argument.